MobileApplication/MobileApplication/ViewModels/Goal/GoalsViewModel.cs

```diff
@@ -83,5 +111,48 @@ async void OnItemSelected(Goal goal)
 
             await Shell.Current.Navigation.PushAsync(new GoalDetailPage(goal));
         }
+
+        public decimal CalculateProfitPerMonth()
+        {
+            decimal totalProfit = 0;
+            List<string> uniqueMonths = new List<string>();
+            foreach (Transaction transaction in _transactions)
+            {
+                string date = transaction.TrTime.ToString("yyyyMM");
+                if (!uniqueMonths.Contains(date))
+                {
+                    uniqueMonths.Add(date);
+                }
+                totalProfit += transaction.Amount;
+
+            }
```

Comment:
This could be rewritten with a linq using distinct selection. See: [Select distinct using linq](https://stackoverflow.com/a/19406319)